framework/src/userobject/ShapeElementUserObject.C

```diff
@@ -0,0 +1,32 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "ShapeElementUserObject.h"
+
+template<>
+InputParameters validParams<ShapeElementUserObject>()
+{
+  InputParameters params = validParams<ElementUserObject>();
+  return params;
+}
+
+ShapeElementUserObject::ShapeElementUserObject(const InputParameters & parameters) :
+    ElementUserObject(parameters),
+    _phi(_assembly.phi()),
```

Comment:
This is the same thing that KernelBase does, and in Kernels we also utilize the _phis as derivatives of the coupled variables in computeOffDiagJacobian. What is Kernel doing differently so that it is not an issue there?